titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/service/ScaleDownEvaluator.java

```diff
@@ -177,6 +183,36 @@ private static List<ServiceJobTask> selectTasksToTerminateInEquivalenceGroup(Lis
         return tasksToKill;
     }
 
+    private static List<ServiceJobTask> selectTasksToTerminateInEquivalenceGroupFavorBinPacking(List<ServiceJobTask> allTasks, int targetTerminateCount, TitusRuntime titusRuntime) {
+        List<ServiceJobTask> tasksToKill = new ArrayList<>();
+        Region region = new Region(allTasks);
+
+        while (tasksToKill.size() < targetTerminateCount && region.hasMoreTasks()) {
+            // Step 1: select smallest groups of tasks to scale down
+            // smallest number of tasks on an agent in any zone
+            int smallestGroup = region.getSmallestTaskGroup();
+
+            // Step 2: kill tasks from the smallest groups, trying to maintain zone balancing
+            // There is increased likelihood that this way we will give agents with a smaller number of tasks
+            // a chance of being scaled-down.
+            boolean hasMore = true;
+            while (tasksToKill.size() < targetTerminateCount && hasMore) {
+                Optional<Zone> zoneOpt = region.getSmallestZoneWithTaskGroupSize(smallestGroup);
+                if (hasMore = zoneOpt.isPresent()) {
```

Comment:
This this `=` correct? The if statement just checks assignment? or is it checking the .isPresent?